packages/suite/src/services/coingecko.ts

```diff
@@ -28,41 +29,76 @@ const fetchCoinGecko = async (url: string) => {
         throw new FiatRatesFetchError(`Failed to fetch: ${url}`);
     }
 };
+
+const getTickerConfig = (ticker: TickerId) => {
+    const config = FIAT_CONFIG.tickers.find(t =>
+        ticker.tokenAddress ? t.symbol === ticker.mainNetworkSymbol : t.symbol === ticker.symbol,
+    );
+
+    if (!config) {
+        console.error('buildCoinUrl: cannot find ticker config for ', ticker);
+    }
+    return config;
+};
+
 /**
- * Returns an array with coins supported by CoinGecko API
+ * Build coinUrl using defined coin ids
  *
- * @returns {Promise<any>}
+ * @param {TickerId} ticker
+ * @returns
  */
-export const fetchCoinList = (): Promise<any> => {
-    const url = `${COINGECKO_API_BASE_URL}/coins/list`;
-    return fetchCoinGecko(url);
+const buildCoinUrl = (ticker: TickerId) => {
+    // for token find its main network
+    const config = FIAT_CONFIG.tickers.find(t =>
+        ticker.tokenAddress ? t.symbol === ticker.mainNetworkSymbol : t.symbol === ticker.symbol,
+    );
+
+    if (!config) {
+        console.error('buildCoinUrl: cannot find ticker config for ', ticker);
+        return null;
+    }
+
+    return `${COINGECKO_API_BASE_URL}/coins/${config.coingeckoId}`;
 };
 
 /**
- * Build coinUrl from the `ticker.coinData` if available.
- * Otherwise fetch coin ID from CoinGecko and use that.
+ * Returns the current rate for a given token fetched from CoinGecko API.
+ * Returns null if main network for the token is not ethereum.
+ * Supports only tokens on ethereum.
  *
- * @param {FiatTicker} ticker
+ * @param {TickerId} ticker
  * @returns
  */
-const buildCoinUrl = (ticker: FiatTicker) => {
-    let coinUrl: string | null = null;
-    const { coinData } = ticker;
-    if (coinData) {
-        // fetch coin id from coingecko and use it to build URL for fetching rates
-        coinUrl = `${COINGECKO_API_BASE_URL}/coins/${coinData.id}`;
+export const fetchCurrentTokenFiatRates = async (ticker: TickerId) => {
+    if (!ticker.tokenAddress) return null;
+
+    const networkTickerConfig = getTickerConfig(ticker);
+    if (!networkTickerConfig || networkTickerConfig?.coingeckoId !== 'ethereum') {
+        console.warn('fetchCurrentTokenFiatRates: This API supports only ethereum', ticker);
+        return null;
     }
-    return coinUrl;
+
+    const url = `${COINGECKO_API_BASE_URL}/simple/token_price/${
+        networkTickerConfig.coingeckoId
+    }?contract_addresses=${ticker.tokenAddress}&vs_currencies=${FIAT_CONFIG.currencies.join(',')}`;
+
+    const rates = await fetchCoinGecko(url);
+    if (!rates) return null;
+
+    return {
+        ts: new Date().getTime() / 1000,
```

Comment:
a total nitpick:
We could use a helper fn to create timestamps. I also saw `getUnixTime(new Date()) * 1000` in `suite/src/actions/wallet/fiatRatesActions.ts`

Reply:
hmmm `getTime` basically already returns unix timestamp, but in milliseconds. Importing another util function and using it versus manually dividing by 1000 seems equal to me 😁 Maybe we could do the change in fiatRatesActions and switch `getUnixTime(new Date()) * 1000` to `new Date().getTime()`? Both should do the same right? I liked date-fns too much it seems 😁 

I use to have problems with such util functions, I never remember if the param should be in milliseconds (what JS does) or in seconds. And because it can't be type-checked it already caused by a headache few times 😁